record/hop.go

```diff
@@ -16,6 +16,10 @@ const (
 	// NextHopOnionType is the type used in the onion to reference the ID
 	// of the next hop.
 	NextHopOnionType tlv.Type = 6
+
+	// MetadataOnionType is the type used in the onion for the payment
+	// metadata.
+	MetadataOnionType tlv.Type = 10
```

Comment:
@Roasbeef, I've created a draft PR to evaluate touch points. Do you want to go for this?

Reply:
Makes a ton of sense IMO, super useful for the AMP recurring case as well. Could also bundle it w/ the `pubkey` based routing stuff as well...

Reply:
Why is it useful for the AMP recurring case actually? The sending of metadata I get. This could be something like a donation comment. But what would be the purpose of embedding metadata in the payment request? Recurring payment senders would all pass through that same metadata, but what does the receiver learn from this that they didn't already know?